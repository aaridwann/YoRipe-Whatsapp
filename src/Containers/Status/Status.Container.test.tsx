import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';

import StatusContainer from './Status.Container';

const mockGenerateFix = [
    {
        date: '01.00',
        image: 'Image01.jpg',
        latestChat: 'hello',
        name: 'Joy',
        notification: 0,
    },
    {
        date: '01.00',
        image: 'Image02.jpg',
        latestChat: 'hello',
        name: 'Joy',
        notification: 0,
    },
    {
        date: '01.00',
        image: 'Image03.jpg',
        latestChat: 'hello',
        name: 'Joy',
        notification: 0,
    },
];
beforeEach(() => {
    jest.clearAllMocks();
});

jest.mock('../ChatList/ChatList.Utils', () => ({
    mapUserGenerator: jest.fn().mockReturnValue(mockGenerateFix),
}));

describe('StatusContainer', () => {
    describe('SnapTest', () => {
        it('Should render correct value', () => {
            expect(render(<StatusContainer />)).toMatchSnapshot();
        });
    });

    it('Should called useState when press image', () => {
        const useState = jest.spyOn(React, 'useState');

        const { getAllByLabelText } = render(<StatusContainer />);

        const ImageButton = getAllByLabelText('imageLabel')[0];

        fireEvent.press(ImageButton);

        expect(useState).toHaveBeenCalled();
    });

    it('Should render Modal when image pressed', async () => {
        const { getAllByLabelText, getByLabelText } = render(
            <StatusContainer />
        );

        const ImageButton = getAllByLabelText('imageLabel')[0];
        const modal = getByLabelText('Modal');

        fireEvent.press(ImageButton);

        expect(modal).toBeTruthy();
    });
});
