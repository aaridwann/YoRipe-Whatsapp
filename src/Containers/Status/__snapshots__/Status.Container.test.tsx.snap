// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusContainer SnapTest Should render correct value 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginHorizontal": 20,
          "marginVertical": 16,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <View
          accessibilityLabel="imageLabel"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Image
            source={
              {
                "uri": "https://i.pinimg.com/564x/44/a9/0d/44a90dfa9f4e8efe32b0070d49f9de30.jpg",
              }
            }
            style={
              {
                "borderColor": "#c9c9c9",
                "borderRadius": 50,
                "borderWidth": 2,
                "height": 100,
                "width": 100,
              }
            }
          />
        </View>
        <View
          style={
            {
              "marginHorizontal": 20,
            }
          }
        >
          <Text
            style={
              {
                "fontWeight": "700",
              }
            }
          >
            Ridwan
          </Text>
          <Text
            style={
              {
                "color": "#a0a0a0",
                "fontSize": 20,
                "fontWeight": "700",
                "marginTop": 8,
              }
            }
          >
            Today Today 01.11
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        IonIcons
      </View>
    </View>
    <Text
      style={
        {
          "backgroundColor": "#e9e9e9",
          "color": "#075e55",
          "fontWeight": "600",
          "padding": 20,
        }
      }
    >
      Recent Update
    </Text>
    <RCTScrollView
      ItemSeparatorComponent={[Function]}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View />
    </RCTScrollView>
    <Text
      style={
        {
          "backgroundColor": "#e9e9e9",
          "color": "#075e55",
          "fontWeight": "600",
          "padding": 20,
        }
      }
    >
      Viewed updates
    </Text>
    <RCTScrollView
      ItemSeparatorComponent={[Function]}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View />
    </RCTScrollView>
    <Modal
      accessibilityLabel="Modal"
      animationType="slide"
      hardwareAccelerated={true}
      onDismiss={[Function]}
      onRequestClose={[Function]}
      transparent={true}
      visible={false}
    />
  </View>
</RCTScrollView>
`;
